src/NewTools-Debugger-Tests/StDebuggerTest.class.st

```diff
@@ -383,6 +390,38 @@ StDebuggerTest >> testCopyStackToClipboard [
 				 session interruptedContext shortDebugStackOn: s ])
 ]
 
+{ #category : #tests }
+StDebuggerTest >> testCreateMissingMethodWithUnfilteredStackUpdatesCodePane [
+
+	| dnuMessage |
+	StDebuggerActionModel shouldFilterStack: false.
+	dbg := StTestDebuggerProvider new debuggerWithDNUContext.
+	dbg
+		application: dbg class currentApplication;
+		initialize.
+	self assert:
+		dbg debuggerActionModel isInterruptedContextDoesNotUnderstand.
+	self
+		assert: dbg interruptedContext receiver class
+		identicalTo: StTestDebuggerProvider.
+	dnuMessage := dbg debuggerActionModel contextPredicate exception
+		              message.
+
+	self assert: dnuMessage selector identicalTo: #foobar.
+
+	"stack is unfiltered. So the text shown should be the code from the DNU"
+	self assert: (dbg code text beginsWith: 'doesNotUnderstand:').
+
+	"we create the missing method #foobar"
+	dbg createMissingMethodFor: dnuMessage in: StTestDebuggerProvider.
+
+	"code must have been updated with the new method"
+	self
+		assert: dbg interruptedContext compiledCode selector
+		identicalTo: #foobar.
+	self assert: (dbg code text beginsWith: 'foobar')
+]
```

Comment:
Should we also test that the *new current context* method is the method `foobar`?

Reply:
And maybe also that the correct text portion has been highlighted?

Reply:
> Should we also test that the _new current context_ method is the method `foobar`?

I've tested only the method selector and not the method itself because I cannot reference a method that doesn't exist. How else could I check that? Maybe with something like:
```Smalltalk
self assert: dbg interruptedContext compiledCode identicalTo: (StTestDebuggerProvider methodDictionary at: #foobar)
```

Like that, I could check the entire source code so it could be better.

> And maybe also that the correct text portion has been highlighted?

Maybe we could test that but that's not what I wanted to test